src/main/common/utils.h

```diff
@@ -90,3 +90,9 @@ void * memcpy_fn ( void * destination, const void * source, size_t num ) asm("me
 
 
 #endif
+
+#if __GNUC__ >= 7
+#define FALLTHROUGH __attribute__ ((fallthrough)
+#else
+#define FALLTHROUGH
```

Comment:
This may eventually bite us. Imagine this code (with a deliberately missing semicolon after a function all:
```
functionCall()
FALLTORUGH;
```
On GCC < 7 it will compile correctly without yielding a syntax error.

Reply:
Ooops, that's one of those mistakes that are totally obvious in hindsight 🤦‍♂️. I'll change the macro to use `do {} while(0)` on GCC < 7, so it won't accidentally "fix" a missing semicolon.